compiler-plugin/src/main/kotlin/arrow/meta/quotes/element/WhenEntry.kt

```diff
@@ -0,0 +1,44 @@
+package arrow.meta.quotes.element
+
+import arrow.meta.quotes.Scope
+import arrow.meta.quotes.ScopedList
+import org.jetbrains.kotlin.psi.KtExpression
+import org.jetbrains.kotlin.psi.KtWhenCondition
+import org.jetbrains.kotlin.psi.KtWhenEntry
+
+/**
+ * <code> """ ${if (!isElse) """$conditions -> $expression""" else  """else -> $expression"""  } """.whenEntry </code>
+ *
+ * A template destructuring [Scope] for a [KtWhenEntry].
+ *
+ * ```kotlin:ank:silent
+ * import arrow.meta.Meta
+ * import arrow.meta.Plugin
+ * import arrow.meta.invoke
+ * import arrow.meta.quotes.Transform
+ * import arrow.meta.quotes.whenEntry
+ *
+ * val Meta.reformatWhenEntry: Plugin
+ *  get() =
+ *   "ReformatWhenEntry" {
+ *    meta(
+ *     whenEntry({ true }) { e ->
+ *      Transform.replace(
+ *       replacing = e,
+ *       newDeclaration = """ ${if (!isElse) """$conditions -> $expression""" else  """else -> $expression"""  } """.whenEntry
+ *      )
+ *     }
+ *    )
+ *   }
+ * ```
+ */
+
+class WhenEntry(
+  override val value: KtWhenEntry?,
+  val conditions: ScopedList<KtWhenCondition> = ScopedList(
+    separator = " && ",
+    value = value?.conditions?.toList() ?: listOf()
+  ),
```

Comment:
For things like this and also the catch clauses list, I'm wondering if we should be putting the logic in the string transformation since it can't exactly be done via scoped list

Reply:
yes, it can be done like that but the separator can be any other binary expression that should come from the model somwhere